src/js/angular/guides/tour-lib-services/shepherd.service.js

```diff
@@ -0,0 +1,543 @@
+     * @returns the last step.
+     * @private
+     */
+    this._getLastStep = (guide, stepsDescriptions) => {
+        return this._toGuideStep(guide, stepsDescriptions[stepsDescriptions.length - 2], stepsDescriptions[stepsDescriptions.length - 1]);
+    }
+
+    /**
+     * Converts <code>currentStepDescription</code> to Shepherd Tour step.
+     * @param guide - the guide.
+     * @param previousStepDescription - the previous step description.
+     * @param currentStepDescription - processed step description.
+     * @param nexStepDescription - the next step description.
+     * @returns created step.
+     * @private
+     */
+    this._toGuideStep = (guide, previousStepDescription, currentStepDescription, nexStepDescription) => {
+        const step = this._toBaseGuideStep($translate, currentStepDescription, () => this._updateLocalStorage());
+        const buttons = [];
+        buttons.push(this._getPreviousButton(guide, previousStepDescription, currentStepDescription));
+        buttons.push(this._getNextButton(guide, currentStepDescription, nexStepDescription));
+        buttons.push(this._getPauseButton(guide));
+        step.buttons = buttons;
+        return step;
+    }
+
+    this._getPauseButton = (guide) => {
+        return this._getButton($translate.instant('guide.button.pause'), () => this._pauseGuide(guide), true);
+    }
+
+    /**
+     * Pauses current ran guide.
+     */
+    this._pauseGuide = (guide) => {
+        guide.hide();
+        this._getStepWhichCanBePaused(Shepherd.activeTour);
+        LocalStorageAdapter.set(GUIDE_PAUSE, true);
+        if (this.onPause()) {
+            this.onPause();
+        }
+    }
+
+    this._getStepWhichCanBePaused = (tour) => {
+        const currentStep = tour.getCurrentStep();
+        if (currentStep) {
+            if (currentStep.options.canBePaused) {
+                return currentStep;
+            }
+            tour.back();
+            tour.hide();
+            const previousStep = tour.getCurrentStep();
+            // checks if ids are equal this mean first step is reached.
+            if (currentStep.id === previousStep.id) {
+                return currentStep;
+            } else {
+                this._getStepWhichCanBePaused(tour);
+            }
+        }
+    }
+
+    /**
+     * Creates a previous button.
+     * @param guide - the guide.
+     * @param previousStepDescription - previous step description.
+     * @param currentStepDescription - processed step description.
+     * @returns created step.
+     * @private
+     */
+    this._getPreviousButton = (guide, previousStepDescription, currentStepDescription) => {
+        const text = $translate.instant('guide.button.previous');
+        const action = this._getPreviousButtonAction(guide, previousStepDescription, currentStepDescription);
+        const button = this._getButton(text, action);
+        button.disabled = !previousStepDescription;
```

Comment:
Not critical, but probably it would be better if the button is not added at all instead of disable it IMO

Reply:
done